asaps/models.py

```diff
@@ -25,36 +27,55 @@ def __init__(self, secfile):
         authclient.authorize()
         self.authclient = authclient
 
-    def getrecord(self, uri):
+    def getrec(self, uri):
         """Retrieve an individual record."""
-        record = self.authclient.get(uri).json()
+        rec = self.authclient.get(uri).json()
         print(uri)
-        recType = record['jsonmodel_type']
-
-        # record type
+        recType = rec['jsonmodel_type']
         if recType == 'resource':
-            fields = [f(field) for field in attr.fields(Resource)]
-            kwargs = {k: v for k, v in record.items() if k in fields}
-            kwargs['jsonstr'] = record
-            kwargs['updjsonstr'] = record
-            rec = Resource(**kwargs)
+            rec = self.popinst(Resource, rec)
         elif recType == 'accession':
-            fields = [f(field) for field in attr.fields(Accession)]
-            kwargs = {k: v for k, v in record.items() if k in fields}
-            kwargs['jsonstr'] = record
-            kwargs['updjsonstr'] = record
-            rec = Resource(**kwargs)
+            rec = self.popinst(Accession, rec)
         elif recType == 'archival_object':
-            fields = [f(field) for field in attr.fields(ArchivalObject)]
-            kwargs = {k: v for k, v in record.items() if k in fields}
-            kwargs['jsonstr'] = record
-            kwargs['updjsonstr'] = record
-            rec = Resource(**kwargs)
+            rec = self.popinst(ArchivalObject, rec)
         else:
-            print('Invalid record type')  # likely better ways of handling this
+            print('Invalid record type')
             exit()
         return rec
 
+    def stringsearch(self, string, repoid, rectype):
+        """Search for a string across a particular record type."""
+        endpoint = ('repositories/' + str(repoid) + '/search?q="'
```

Comment:
I would get in the habit of using either `str.format()` or the new (as of 3.6) f-string syntax for formatting strings. Both are acceptable and which you use will mostly just depend on how readable you think one is over the other in each case. It is much more flexible and usually more readable than just string concatenation. I tend to prefer f-strings over format these days.

Example with `format`:
```python
endpoint = 'repositories/{}/search?q="{}"&page_size=100&type[]={}'.format(repoid, string, rectype)
```
and with f-string:
```python
endpoint = f'repositories/{repoid}/search?q="{string}"&page_size=100&type[]={rectype}'
```

Reply:
This is awesome